internal/utils/utils.go

```diff
@@ -54,7 +55,11 @@ func DownloadFile(url string) (string, error) {
 	}
 	defer tempFile.Close()
 
-	filepathFull := tempFile.Name()
+	if downloadPath != "" {
+		filepathFull = downloadPath
+	} else {
```

Comment:
Could you move https://github.com/elastic/e2e-testing/pull/1984/files#diff-a7cadd8b7616e655fd258138320b33eb96f7cce8e8663ec92651f49ce07df000R45-R46 right here, covered by the else branch? We will be creating the temp file if and only if the downloadPath is not empty